types/mempool/priority_nonce.go

```diff
@@ -287,6 +287,7 @@ func (mp *PriorityNonceMempool[C]) Insert(ctx context.Context, tx sdk.Tx) error
 
 func (i *PriorityNonceIterator[C]) iteratePriority() Iterator {
 	// beginning of priority iteration
+	i.mempool.mtx.Lock()
```

Comment:
Upstream bug?

Reply:
yea, but the fix seems tricky,  instead of access priorityIndex/priorityNode directly, I'm thinking wrap sth like
```
type atomicPriorityIndex struct {
	atomic.RWMutex
	priorityIndex *ListIndex
	priorityNode *ListNode
}

func (i *atomicPriorityIndex) SetIndex(xxx) {
 	// lock
	i.priorityIndex.Set(xxx)
}

func (i *atomicPriorityIndex) NextNode() *ListNode {
	// lock
	return i.priorityNode.Next()
}
```